backend/src/main/java/org/eclipse/tractusx/puris/backend/supply/logic/service/CustomerSupplyService.java

```diff
@@ -110,17 +112,37 @@ public final List<ReportedCustomerSupply> findAllByFilters(Optional<String> ownM
     }
 
     public boolean validate(ReportedCustomerSupply daysOfSupply) {
-        return 
-            daysOfSupply.getPartner() != null &&
-            daysOfSupply.getMaterial() != null &&
-            daysOfSupply.getDate() != null &&
-            daysOfSupply.getDaysOfSupply() >= 0 &&
-            daysOfSupply.getStockLocationBPNS() != null &&
-            daysOfSupply.getStockLocationBPNA() != null &&
-            daysOfSupply.getPartner() != partnerService.getOwnPartnerEntity() &&
-            daysOfSupply.getPartner().getSites().stream().anyMatch(site -> 
-                site.getBpns().equals(daysOfSupply.getStockLocationBPNS()) && 
-                site.getAddresses().stream().anyMatch(address -> address.getBpna().equals(daysOfSupply.getStockLocationBPNA()))
-            );
+        return validateWithDetails(daysOfSupply).isEmpty();
+    }
+
+    public List<String> validateWithDetails(ReportedCustomerSupply daysOfSupply) {
+        List<String> errors = new ArrayList<>();
+        Partner ownPartnerEntity = partnerService.getOwnPartnerEntity();
+
+        if (daysOfSupply.getMaterial() == null) {
+            errors.add("Missing Material.");
+        }
+        if (daysOfSupply.getPartner() == null) {
+            errors.add("Missing Partner.");
+        }
+        if (daysOfSupply.getDate() == null) {
+            errors.add("Missing date.");
+        }
+        if (daysOfSupply.getStockLocationBPNS() == null) {
+            errors.add("Missing stock location BPNS.");
+        }
+        if (daysOfSupply.getStockLocationBPNA() == null) {
+            errors.add("Missing stock location BPNA.");
+        }
+        if (daysOfSupply.getPartner().equals(ownPartnerEntity)) {
+            errors.add("Partner cannot be the same entity.");
+        }
+        if (daysOfSupply.getPartner().getSites().stream().noneMatch(site -> 
+                site.getBpns().equals(daysOfSupply.getStockLocationBPNS()) || 
+                site.getAddresses().stream().noneMatch(address -> address.getBpna().equals(daysOfSupply.getStockLocationBPNA()))
+            )) {
+            errors.add("Invalid days of supply: stock location is not valid.");
```

Comment:
```suggestion
            errors.add(String.format("Invalid days of supply: stock location '%s' and or stock address '%s' don't belong to each other or partner '%s'.", daysOfSupply.getStockLocationBPNS(), daysOfSupply.getStockLocationBPNA(), daysOfSupply.getPartner().getBpnl()));
```

Please consider a more detailed error message to improve understandability.